### PR TITLE
EntryHash and AspectHash

### PR DESCRIPTION
**Pull request by ddd-mtl**

### CODE REVIEW

File: crates/lib3h_protocol/src/types.rs [hunk 1]
```diff
@@ -43,3 +43,89 @@ impl SpaceHash {
         SpaceHash(HashString::new())
     }
 }
+
+//--------------------------------------------------------------------------------------------------
+// EntryHash: newtype for HashString
+//--------------------------------------------------------------------------------------------------
+
+#[derive(PartialOrd, PartialEq, Eq, Ord, Clone, Debug, Serialize, Deserialize, Default, Hash)]
+pub struct EntryHash(HashString);
```
Comment (neonphog) [inline crates/lib3h_protocol/src/types.rs]: Can you also do Shrinkwrap and shrinkwrap(mutable) a'la https://github.com/holochain/lib3h/blob/develop/crates/zombie_actor/src/lib.rs#L22-L23 so we get AsRef and friends?

File: crates/lib3h/src/dht/mod.rs [hunk 2]
```diff
@@ -86,8 +86,8 @@ pub mod tests {
 
     #[allow(non_snake_case)]
     fn create_EntryData(
-        entry_address: &Address,
-        aspect_address: &Address,
+        entry_address: &EntryHash,
+        aspect_address: &AspectHash,
```
Comment (neonphog) [inline crates/lib3h/src/dht/mod.rs]: do we want to rename the fields to entry_hash && aspect_hash? I'm not sure I have an opinion either way, just asking the question.

Comment (ddd-mtl) [reply crates/lib3h/src/dht/mod.rs]: no we dont. No point in specifying type in variable name. Its used as an address so name is fine

File: crates/lib3h_protocol/src/types.rs (hunk 1 above)
Comment (ddd-mtl) [reply crates/lib3h_protocol/src/types.rs]: shrinkwrap(mutable) doesnt work as HashString would need to derive it also I believe.
Don't see the need anyway as there is no need to mutate hashes, no?

**Pull request merged**